src/Resolvers/Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver/WorkloadSdkResolver.cs

```diff
@@ -63,14 +67,17 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
                 resolverContext.State = cachedState;
             }
 
-            string userProfileDir = CliFolderPathCalculatorCore.GetDotnetUserProfileFolderPath();
-            var result = cachedState.WorkloadResolver.Resolve(sdkReference.Name, cachedState.DotnetRootPath, cachedState.SdkVersion, userProfileDir, cachedState.GlobalJsonPath);
-
+            string? userProfileDir = CliFolderPathCalculatorCore.GetDotnetUserProfileFolderPath();
+            ResolutionResult? result = null;
+            if (cachedState.DotnetRootPath is not null && cachedState.SdkVersion is not null)
+            {
+                result = cachedState.WorkloadResolver?.Resolve(sdkReference.Name, cachedState.DotnetRootPath, cachedState.SdkVersion, userProfileDir, cachedState.GlobalJsonPath);
+            }
 
-            return result.ToSdkResult(sdkReference, factory);
+            return result?.ToSdkResult(sdkReference, factory);
```

Comment:
This should maybe return some kind of failed result if it doesn't get a real one.

Reply:
So, I'm not adding/changing logic here in this PR if I can avoid it. Originally, using null-forgiving operator accomplished that, but for this situation and no longer using that operator, the properties `DotnetRootPath`, `SdkVersion`, and `WorkloadResolver` are not guaranteed to be non-null without redesigning the `CachedState` class for those properties. Therefore, this method cannot guarantee a result.

Keep in mind, this PR doesn't intend on enhancing the code logic in any way if possible. It is only to make it compatible with nullability, not redesign the code around nullability. That would be a major project and take a long time.